types/schema.d.ts

```diff
@@ -26,6 +26,7 @@ export interface FastifySchemaValidationError {
 }
 
 export interface FastifyValidationResult {
+  (this: any, data: any): unknown;
```

Comment:
Why using unknown? I suppose this should be the interface of a `validatorCompiler` that returns a function that returns what is currently implemented as `FastifyValidationResult`. Am I wrong?

Reply:
Reading the suggestion at https://github.com/fastify/fastify/pull/2807#pullrequestreview-572409265, the function should return a boolean.

Reply:
It should not be returning boolean as fastify accept other returing value such as 
```
{ error: Error, value: any }
```
In here, returning `unknown` provide the highest flexibility for the user to pass in the function.

Reply:
You can look at the implementation here.
https://github.com/fastify/fastify/blob/master/lib/validation.js#L70-L76

Reply:
Maybe we just change the type back to this one? @VigneshMurugan 
```suggestion
  (data: any): boolean | PromiseLike<any> | { error?: Error, value?: any }
```

Reply:
I initially had it as per your suggestion but was unsure if this is an acceptable type hence went for unknown. If nobody has an objection i can go for your suggestion !

CC: @mcollina @fox1t @Ethan-Arrowood

Reply:
Go for it!